examples/reshape/app.tsx

```diff
@@ -0,0 +1,74 @@
+/** @jsx h */
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createQuerySuggestionsPlugin } from '@algolia/autocomplete-plugin-query-suggestions';
+import { createLocalStorageRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+import { h, Fragment } from 'preact';
+import { pipe } from 'ramda';
+
+import '@algolia/autocomplete-theme-classic';
+
+import { groupBy, limit, uniqBy } from './functions';
+import { productsPlugin } from './productsPlugin';
+import { searchClient } from './searchClient';
+
+const recentSearchesPlugin = createLocalStorageRecentSearchesPlugin({
+  key: 'search',
+  limit: 10,
+});
+const querySuggestionsPlugin = createQuerySuggestionsPlugin({
+  searchClient,
+  indexName: 'instant_search_demo_query_suggestions',
+  getSearchParams() {
+    return {
+      hitsPerPage: 10,
+    };
+  },
+});
+
+const combineSuggestions = pipe(
+  uniqBy(({ source, item }) =>
+    source.sourceId === 'querySuggestionsPlugin' ? item.query : item.label
+  ),
+  limit(4)
+);
+const groupByCategory = groupBy((hit) => hit.categories[0], {
+  getSource({ name, items }) {
+    return {
+      getItems() {
+        return items.slice(0, 3);
+      },
+      templates: {
+        header() {
+          return (
+            <Fragment>
+              <span className="aa-SourceHeaderTitle">{name}</span>
+              <div className="aa-SourceHeaderLine" />
+            </Fragment>
+          );
+        },
+      },
+    };
+  },
+});
+
+autocomplete({
+  container: '#autocomplete',
+  placeholder: 'Search',
+  debug: true,
+  openOnFocus: true,
+  plugins: [recentSearchesPlugin, querySuggestionsPlugin, productsPlugin],
```

Comment:
why is products a plugin and not a regular source? could indicate more "regular" usage?

Reply:
this gives me the idea if longer term plugins and sources maybe could be merged into a single argument? plugins could have no source, and it would allow you to order before using reshape already

Reply:
> why is products a plugin and not a regular source? could indicate more "regular" usage?

I wanted to focus the example on the reshape API, and not pollute the file with a huge source.

Reply:
> this gives me the idea if longer term plugins and sources maybe could be merged into a single argument? plugins could have no source, and it would allow you to order before using reshape already


You mean something like this?

```ts
autocomplete({
  // ...
  getSources() {
    return [
      recentSearchesPlugin,
      querySuggestionsPlugin,
      {
        // custom source
      }
    ]
  }
})
```

Reply:
I was more thinking of `plugins: [somePlugin, { getSources() { customStuff }}`, maybe that way you don't really need the callback form (although that limits some use cases for conditional requests, in which case your suggestion fits better)

I was just wondering if we really need to distinguish between the two